tests/unit/challenges/test_aws_utils.py

```diff
@@ -0,0 +1,511 @@
+                name="Challenge Phase 2",
+                description="Description for Challenge Phase",
+                start_date=timezone.now() - timedelta(days=2),
+                end_date=timezone.now() + timedelta(days=1),
+                challenge=self.challenge2,
+            )
+
+        self.client.force_authenticate(user=self.user)
+        self.ecs_client = boto3.client("ecs", region_name=os.environ.get("AWS_DEFAULT_REGION", "us-east-1"), aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"), aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),)
+
+
+class TestClientTokenGen(TestCase):
+    def test_client_token_gen(self):
+        response = aws_utils.client_token_generator()
+        self.assertTrue(all(ord(c) < 128 for c in response))
+
+
+@mock_ecs
+class TestRegisterTaskDefinition(BaseTestClass):
+    def setUp(self):
+        super(TestRegisterTaskDefinition, self).setUp()
+
+    def test_register_task_def_by_challenge_pk_for_new_challenge(self):
+        queue_name = self.challenge.queue
+        expected_arn = "arn:aws:ecs:us-east-1:012345678910:task-definition/test_queue:1"
+        expected_status_code = HTTPStatus.OK
+        response = aws_utils.register_task_def_by_challenge_pk(self.ecs_client, queue_name, self.challenge)
+        status_code = response["ResponseMetadata"]["HTTPStatusCode"]
+        arn = response["taskDefinition"]["taskDefinitionArn"]
+        self.assertEqual(arn, expected_arn)
+        self.assertEqual(status_code, expected_status_code)
+
+    def test_register_task_def_by_challenge_pk_for_existing_service(self):
+        queue_name = self.challenge2.queue
+        aws_utils.register_task_def_by_challenge_pk(self.ecs_client, queue_name, self.challenge2)
+        expected_status_code = None
+        expected_error_message = "Error. Task definition already registered for challenge {}.".format(self.challenge2.pk)
+        response = aws_utils.register_task_def_by_challenge_pk(self.ecs_client, queue_name, self.challenge2)
+        status_code = response["ResponseMetadata"]["HTTPStatusCode"]
+        error_message = response["Error"]
+        self.assertEqual(expected_error_message, error_message)
+        self.assertEqual(status_code, expected_status_code)
+
+    def test_register_task_def_by_challenge_pk_for_empty_task_role_arn(self):
+        aws_utils.TASK_ROLE_ARN = ""
+        queue_name = self.challenge2.queue
+        expected_status_code = None
+        expected_error_message = "Please ensure that the TASK_ROLE_ARN & TASK_EXECUTION_ROLE_ARN are appropriately passed as environment varibles."
+        response = aws_utils.register_task_def_by_challenge_pk(self.ecs_client, queue_name, self.challenge2)
+        status_code = response["ResponseMetadata"]["HTTPStatusCode"]
+        error_message = response["Error"]
+        self.assertEqual(expected_error_message, error_message)
+        self.assertEqual(status_code, expected_status_code)
```

Comment:
@KhalidRmb @RishabhJain2018  do we need tests for missing challenges also?

Reply:
This method is only called indirectly through the admin actions, so in no scenario will it be called on  missing challenges. So, I didn't put any such test case.